### DISCUSSION TITLE
FR: Let me define input masks on plain text fields

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/cms#8805] [Ideas]
### Description
I would love to be able to apply an input mask to plain text fields in order to help content creators stay consistent with things like phone numbers, email addresses, URLs, etc. Ideally, I could define these with configuration files, similar to how Redactor can have configurations.

The [imaskjs](https://github.com/uNmAnNeR/imaskjs) library looks well maintained, is appropriately licensed, and has a Vue plugin available to assist in the dashboard's transition to Vue in later versions of Craft.

Bonus points for:
- Let me choose whether to save the masked value or the raw value
- Let me use this on number fields as well
- Let me write complex [configurations](https://imask.js.org/guide.html#masked-base)
- Come up with a bespoke solution that can validate on both the frontend and backend

### Additional info
- Craft version: 3.4.x

**Comment by piotrpog** [comment]
Very good idea!
